CocktailsOrder/CocktailsOrder/Networking/CocktailModelData.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  CocktailModelData.swift
+//  CocktailsOrder
+//
+//  Created by Cosma Dragos on 27.07.2022.
+//
+
+import Foundation
+import Combine
+
+final class CocktailModelData: ObservableObject {
+    @Published var cocktails: [Cocktail] = []
+    @Published var likedCocktails: [Cocktail] = []
+    
+    private var bag = Set<AnyCancellable>()
+
+    init() {
+        self.fetchCocktails()
+    }
+    
+    func fetchCocktails() {
+        let cocktailsUrlString = "https://www.thecocktaildb.com/api/json/v1/1/filter.php?a=Alcoholic"
+        
+        if let url = URL(string: cocktailsUrlString) {
+            URLSession.shared
+                .dataTaskPublisher(for: url)
+                .receive(on: DispatchQueue.main)
+                .map(\.data)
+                .decode(type: CocktailsResponse.self, decoder: JSONDecoder())
+                .sink { res in
+                    switch res {
+                    case .failure(let error):
+                        print(error)
+                    default: break
+                    }
+                } receiveValue: { cocktailsData in
+                    self.cocktails = cocktailsData.drinks ?? []
+                }
+                .store(in: &bag)
```

Comment:
shouldn't the NetworkServiceable async call method be used here? instead of creating a URLSession in the VM? :D

Reply:
This model was used for the first version of the Networking(not the one that uses async await)